components/Primitives/src/StaggeredLayout/StaggeredLayout.cs

```diff
@@ -142,16 +158,35 @@ protected override Size MeasureOverride(VirtualizingLayoutContext context, Size
         double availableHeight = availableSize.Height;
 
         // This ternary prevents the column width from being NaN, which would otherwise cause an exception when measuring item sizes
-        double columnWidth = double.IsNaN(DesiredColumnWidth) ? availableWidth : Math.Min(DesiredColumnWidth, availableWidth);
+        double columnWidth;
+        int numColumns;
+        if (ItemsStretch is StaggeredLayoutItemsStretch.None)
+        {
+            columnWidth = double.IsNaN(DesiredColumnWidth) ? availableWidth : Math.Min(DesiredColumnWidth, availableWidth);
+            numColumns = Math.Max(1, (int)Math.Floor(availableWidth / state.ColumnWidth));
```

Comment:
🦙 Leaving a marker here for historical record/the future that we had missed the `state.ColumnWidth` here not being the desired value of `columnWidth` calculated above, as `@Poker-sang` called out here: https://github.com/CommunityToolkit/Windows/issues/542#issuecomment-2505215836